core/pkg/runtime/from_config.go

```diff
@@ -218,24 +221,19 @@ func ParseSources(sourcesFlag string) ([]SourceConfig, error) {
 	if err := json.Unmarshal([]byte(sourcesFlag), &syncProvidersParsed); err != nil {
 		return syncProvidersParsed, fmt.Errorf("unable to parse sync providers: %w", err)
 	}
-	for i, sp := range syncProvidersParsed {
+	for _, sp := range syncProvidersParsed {
 		if sp.URI == "" {
 			return syncProvidersParsed, errors.New("sync provider argument parse: uri is a required field")
 		}
 		if sp.Provider == "" {
 			return syncProvidersParsed, errors.New("sync provider argument parse: provider is a required field")
 		}
-		switch uriB := []byte(sp.URI); {
```

Comment:
this switch should be added back in, it prevents errors if a developer was to use a prefix. While this isn't expected behaviour it shouldn't break if they do.

Reply:
it should also be expanded to include the grpc(s) regex, which would prevent this from being a breaking change

Reply:
I disagree. 

If we allow such a fall-back mechanism, then we again have mixed usage of source configurations. And those lead to confusion (both users and developers). Not to mention documentation and potential backward compatibility issues.

My take is - we must be strict on what we support and make a clear separation between  `--uri` vs `--sources` based configurations. And document the correct behavior. Otherwise, we will end up with discussions like this [1]. 

Regarding the breaking change - It's best to do this now rather delaying further. 

[1] - https://github.com/open-feature/open-feature-operator/pull/408#discussion_r1143085223